extensions/ql-vscode/src/run-queries.ts

```diff
@@ -212,7 +239,28 @@ export class QueryEvaluationInfo {
             })
 
             .catch(err => {
-              void showAndLogWarningMessage(`Failed to generate structured evaluator log summary. Reason: ${err.message}`);
+              void showAndLogWarningMessage(`Failed to generate human-readable structured evaluator log summary. Reason: ${err.message}`);
+            });
+
+          // REVIEW: it seems clunky to have to run this command again just to get the same summary file
+          // in JSON format, but I'm not sure it's worth changing the way it works in the CLI for just
+          // this use case? 
```

Comment:
I think this is fine for now. I do think it's looking like we want a way to specify which types of output to send where in `codeql generate log-summary`. Then we could write a single invocation like `codeql generate log-summary --output "end-table:some-file" --output "summary:some-other-file" --output "human-readable:yet-another-file"`. But having something like that would be a pretty big rewrite of the already quite complex `codeql generate log-summary`.

Reply:
Agreed. Another option I just suggested verbally was to have an `--output ide` or similar, which produces all the different output formats the extension expects at this point, saved in separately named output files. But this can all wait for the future.

Reply:
Sounds good. Just filed https://github.com/github/codeql-core/issues/2480 to track for later and move discussion there!